src/components/ebay-progress-bar-expressive/component.ts

```diff
@@ -0,0 +1,152 @@
+import { AttrString } from "marko/tags-html";
+import type { WithNormalizedProps } from "../../global";
+
+export const messageDurationStandard = 1500;
+export const messageDurationReducedMotionMultiplier = 1.5;
+export const messageFadeInDuration = 833;
+
+interface Message {
+    text: string;
+    duration?: number;
+}
+
+interface ProgressBarExpressiveInput
+    extends Omit<Marko.Input<"div">, `on${string}`> {
+    "a11y-text"?: AttrString;
+    messages?: Marko.RepeatableAttrTag<Message>;
+    size?: "medium" | "large";
+}
+
+export interface Input
+    extends WithNormalizedProps<ProgressBarExpressiveInput> {}
+
+interface State {
+    messages: Message[];
+    messageIsFadingIn: boolean;
+    currentMessageIndex: number;
+    nextMessageIndex: number;
+    isInitialMessage: boolean;
+    useReducedMotion: boolean;
+}
+
+class ProgressBarExpressive extends Marko.Component<Input, State> {
+    declare timeouts: { fadeIn?: NodeJS.Timeout; showMessage?: NodeJS.Timeout };
+    declare fadeInFirstMessage: boolean;
+
+    onCreate(input: Input) {
+        const messages = (input.messages as Message[]) || [];
+        const useReducedMotion =
+            typeof window !== "undefined" &&
+            window.matchMedia("(prefers-reduced-motion: reduce)").matches;
+        // For medium sized text, display the first message immediately
+        this.fadeInFirstMessage = input.size !== "medium";
+        this.state = {
```

Comment:
This makes sense! One of the reasons I was using state here was to cast and store the messages as `Message[]` type, rather than the `RepeatableAttrTag<Message>` type of `input.messages`. There are a few spots where I need to check the # of messages. Would you recommend casting it each time I do a length check - e.g. `(input.messages as Message[])?.length` - or storing message count as a state variable?